pkg/api/types.go

```diff
@@ -341,14 +342,75 @@ type Result struct {
 	BuildInfo BuildInfo
 }
 
-// BuildInfo holds information about a particular step in the build process.
+// BuildInfo contains information about the build process.
 type BuildInfo struct {
+	// Steps contains details about each build step.
+	Steps Steps
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
-	// build, failed.
+	// build failed.
 	FailureReason FailureReason
 }
 
+// Steps is a map of build steps that have been recorded
+type Steps map[StepName]StepInfo
```

Comment:
@jim-minter Thanks, I'll take a look at that once I have clarified what @smarterclayton is suggesting with the tracing library.